server/vault.api.js

```diff
@@ -0,0 +1,128 @@
+const express = require('express');
+const asyncHandler = require('express-async-handler');
+const HomebrewModel = require('./homebrew.model.js').model;
+
+const router = express.Router();
+
+const buildTitleConditions = (title) => {
+    if (!title) return {};
+    return {
+        $text: {
+            $search: title,
+            $caseSensitive: false,
+        },
+    };
+};
+
+const buildAuthorConditions = (author) => {
+    if (!author) return {};
+    return { authors: author };
+};
+
+//"$and": [ {"published": true}, {"$text": { "$search": "titleString", "$caseSensitive": false } }, { "authors" : "authorString"}]
+//is a good example of a query constructed with this function
+
+const handleErrorResponse = (res, error, functionName) => {
+    const status = error.response?.status || 500;
+    const message =
+        status === 503 ? 'Service Unavailable' : 'Internal Server Error';
+
+    console.error(`Error in ${functionName}:`, error);
+
+    return res.status(status).json({
+        errorCode: status.toString(),
+        message: `Error in function ${functionName}: ${message}`,
+    });
+};
+
+const buildBrewsQuery = (legacy, v3) => {
+    const brewsQuery = { published: true };
+    if (legacy === 'true' && v3 === 'true') return { published: true };
+
+    if (legacy === 'true' && v3 !== 'true') {
+        brewsQuery.renderer = 'legacy';
+    } else if (v3 === 'true' && legacy !== 'true') {
+        brewsQuery.renderer = 'V3';
+    }
+
+    return brewsQuery;
+};
+
+const vault = {
+    findBrews: async (req, res) => {
+        try {
```

Comment:
Generally, we have preferred `await.catch()` instead of `try... catch`. Not a huge deal, but may be worth putting it in a consistent format.

See:

https://github.com/naturalcrit/homebrewery/blob/c41116e6c8eadc418b1dcf799dd3a83c74d5bcae/server/homebrew.api.js#L260-L265

Reply:
had to ask the toaster for a difference between the two, alright